### PR TITLE
Remove keyType and valueType from MaxOrMinByState

### PR DESCRIPTION
**Pull request by haozhun**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/MinBy.java [hunk 1]
```diff
@@ -76,12 +76,12 @@ private static InternalAggregationFunction generateAggregation(Type valueType, T
...
+        MaxOrMinByStateSerializer stateSerializer = new MaxOrMinByStateSerializer(valueType, keyType);
         Type intermediateType = stateSerializer.getSerializedType();
 
         List<Type> inputTypes = ImmutableList.of(valueType, keyType);
 
-        MaxOrMinByStateFactory stateFactory = new MaxOrMinByStateFactory(valueType, keyType);
+        MaxOrMinByStateFactory stateFactory = new MaxOrMinByStateFactory();
         AggregationMetadata metadata = new AggregationMetadata(
                 generateAggregationName(NAME, valueType, inputTypes),
                 createInputParameterMetadata(valueType, keyType),
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/MinBy.java]: you need to bindTo here also

**Comment by cberner** [comment]
there are test failures

**Comment by haozhun** [comment]
Will be more careful in making sure pull requests matches local work directory. Please review again.

**Comment by cberner** [comment]
looks good

**Pull request merged**